### PR TITLE
MCKIN-12153 AF#41-Simulated_Chat-Inform assistive technologies of changes in content

### PR DESCRIPTION
**Pull request by nasirhjafri**
## [MCKIN-12153](https://edx-wiki.atlassian.net/browse/MCKIN-12153)

AF#41-Simulated_Chat-Inform assistive technologies of changes in content

### CODE REVIEW

**Comment by nasirhjafri** [comment]
@Agrendalath  @xitij2000  This PR was also part of the tickets sent for review last sprint. Please review.

**Comment by xitij2000** [comment]
@nasirhjafri Just pinging me is sufficient. I think our sprints may be misaligned, I got the tickets for review in the current sprint, and as such they are scheduled for review next sprint, starting this coming monday.

**Comment by nasirhjafri** [comment]
@xitij2000 Reminder: to review this one, thanks.

**Comment by lgp171188** [comment]
@nasirhjafri, I am reviewing this today and hope to have a review pass done before my EOD today.

**Comment by lgp171188** [comment]
@nasirhjafri, I tested it on the Ironwood devstack and found that this PR only adds a highlight (through a dotted border for the whole chat block and not for the individual messages and that too only when the XBlock container has scrollbars to scroll through the chat content. Otherwise, there is no indication of the block or its contents being accessible via the keyboard. I tested this on Firefox 70.0.1.

Should this also be tested in the solutions devstack and the custom LMS frontend? Also can you add some testing instructions on what the expected behaviour should be with the changes in this PR?

**Comment by lgp171188** [comment]
@nasirhjafri, my previous comment was due to misunderstanding the purpose of this PR and the changes made.

Apart from checking that the `aria-live` attribute is present with the `"polite"` value, is there any way to test that this works okay in screen readers?

**Comment by nasirhjafri** [comment]
@lgp171188 No, it only needs to be tested with screen readers.

**Comment by lgp171188** [comment]
@nasirhjafri, I will test with a screen reader today and post an update.

File: setup.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def package_data(pkg, roots):
 
 setup(
     name='chat-xblock',
-    version='0.2.3',
+    version='0.2.4',
```
Comment (Agrendalath) [inline setup.py]: Please bump the version to `0.2.5`.

**Review by Agrendalath — CHANGES_REQUESTED** [review]

**Comment by lgp171188** [comment]
@nasirhjafri, I finally tested this using the "Narrator" screen reader app on Windows 10 and verified that the changes work okay, except for a small issue. All the messages from the bot are read out loud except the last one about the README file. However, it looks like the `<div>` containing that message is not structured differently from the other divs that the app read out aloud. Can you test this and confirm if you see the same issue?

Also can you resolve the conflicts and also bump the version as requested by @Agrendalath?

**Comment by nasirhjafri** [comment]
@lgp171188 Sure lemme test on my side. I've rebased the branch though.

**Comment by lgp171188** [comment]
> @lgp171188 Sure lemme test on my side. I've rebased the branch though.

@nasirhjafri, did you encounter the same issue I reported in an earlier comment?

**Comment by nasirhjafri** [comment]
@lgp171188 It was reading all of the messages. After rebasing from master (changes from https://github.com/open-craft/xblock-chat/pull/35), it started to read messages twice. I've fixed that, rest is good.

**Comment by lgp171188** [comment]
@nasirhjafri, thanks for the update. I am on leave this week and hence will ask a colleague to do a final check and merge this if everything is okay.

**Comment by lgp171188** [comment]
@giovannicimolin, can you do a quick check of the discussion on this PR, check and test it and approve this PR if everything is okay?

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this:
1. Installed this on an Ironwood devstack.
2. Setup the xblock on the demo course.
3. Checked that the chat messages were read correctly by a screen reader :heavy_check_mark: 
   I used `Pericles` chrome browser extension to perform the tests, targeting the reader to read the contents inside the chat box.
- [x] I read through the code
- [x] I checked for accessibility issues: This fixes an accessibility issue.
- [ ] ~~Includes documentation~~ NA

**Comment by giovannicimolin** [comment]
@nasirhjafri I've merged the PR and tagged a new release from the latest `master`. CC @lgp171188

**Comment by nasirhjafri** [comment]
Great, thanks for reviewing @giovannicimolin @lgp171188

**Pull request merged**